app/Jobs/CreateQueryserviceBatchesJob.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Jobs;
+
+use App\EventPageUpdate;
+use App\QsBatch;
+use App\QsCheckpoint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Config;
+
+class CreateQueryserviceBatchesJob extends Job
+{
+    private const NAMESPACE_ITEM = 120;
+    private const NAMESPACE_PROPERTY = 122;
+    private const NAMESPACE_LEXEME = 146;
+
+    private int $entityLimit;
+
+    public function __construct()
+    {
+        $this->entityLimit = Config::get('wbstack.qs_batch_entity_limit');
+    }
+
+    public function handle(): void
+    {
+        DB::transaction(function () {
+            $latestCheckpoint = QsCheckpoint::get();
+
+            [$newEntities, $latestEventId] = $this->getNewEntities($latestCheckpoint);
+            foreach ($newEntities as $wikiId => $entityIdsFromEvents) {
+                try {
+                    $success = $this->tryToAppendEntitesToExistingBatches($entityIdsFromEvents, $wikiId);
+                    if ($success) {
+                        continue;
+                    }
+                    $this->createNewBatches($entityIdsFromEvents, $wikiId);
+                } catch (\Exception $ex) {
+                    Log::error(
+                        'Failed to create batches '.implode(',', $entityIdsFromEvents).' for wiki with id '.$wikiId.': '.$ex->getMessage()
+                    );
+                    $this->fail($ex);
+                }
```

Comment:
Not sure about retrying vs. not failing all wikis on one naughty wiki here.

If we wanted the ideal behavior of:
- one naughty wiki does not block following wikis
- a failing wiki will be retried so we're sure this is not sth intermittent

we'd need to instead fan out each wiki into it's own job.

Not sure.